theatre/studio/src/store/stateEditors.ts

```diff
@@ -657,6 +662,26 @@ namespace stateEditors {
 
             track.keyframes = sorted
           }
+
+          export function mergeKeyframes(
+            p: WithoutSheetInstance<SheetObjectAddress> & {
+              trackId: string
+              keyframes: Array<Keyframe>
+              snappingFunction: SnappingFunction
+            },
+          ) {
+            const track = _getTrack(p)
+            if (!track) return
+
+            const currentKeyframes = current(track.keyframes)
+            const keyframesWithNewIds = p.keyframes.map((kf) => ({
+              ...kf,
+              id: generateKeyframeId(),
```

Comment:
Generating a random ID makes the stateEditor (which is like a redux reducer), not repeatable. It also makes it so that the caller of `mergeKeyframes()` cannot know the IDs of the new keyframes, and so cannot select them or modify them for example.